api/src/main/java/org/openmrs/BaseFormRecordableOpenmrsData.java

```diff
@@ -0,0 +1,104 @@
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.APIException;
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * This is a super class to make a bridge between an OpenMrsObject and is position in the form.
+ * 
+ * @since 1.4
+ */
+@MappedSuperclass
+public abstract class BaseFormRecordableOpenmrsData extends BaseChangeableOpenmrsData implements FormRecordable {
+
+	private static final String FORM_NAMESPACE_PATH_SEPARATOR = "^";
+
+	private static final int FORM_NAMESPACE_PATH_MAX_LENGTH = 255;
+
+	@Column(name = "form_namespace_path")
+	protected String formNamespaceAndPath;
+
+	/**
+	 * @see org.openmrs.FormRecordable#getFormFieldNamespace()
+	 */
+	@Override
+	public String getFormFieldNamespace() {
+		if (StringUtils.isNotBlank(formNamespaceAndPath)) {
+			//Only the path was specified
+			if (formNamespaceAndPath.startsWith(FORM_NAMESPACE_PATH_SEPARATOR)) {
+				return null;
+			}
+			return formNamespaceAndPath.substring(0, formNamespaceAndPath.indexOf(FORM_NAMESPACE_PATH_SEPARATOR));
+		}
+
+		return formNamespaceAndPath;
+	}
+
+	/**
+	 * @see org.openmrs.FormRecordable#getFormFieldPath()
+	 */
+	@Override
+	public String getFormFieldPath() {
+		if (StringUtils.isNotBlank(formNamespaceAndPath)) {
+			//Only the namespace was specified
+			if (formNamespaceAndPath.endsWith(FORM_NAMESPACE_PATH_SEPARATOR)) {
+				return null;
+			}
+			return formNamespaceAndPath.substring(formNamespaceAndPath.indexOf(FORM_NAMESPACE_PATH_SEPARATOR) + 1);
+		}
+
+		return formNamespaceAndPath;
+	}
+
+	/**
+	 * @see org.openmrs.FormRecordable#setFormField(String,String)
+	 */
+	@Override
+	public void setFormField(String namespace, String formFieldPath) {
+		if (namespace == null && formFieldPath == null) {
+			markAsDirty(formNamespaceAndPath, null);
+			formNamespaceAndPath = null;
+			return;
+		}
+
+		String nsAndPathTemp = "";
+		if (StringUtils.isNotBlank(namespace) && StringUtils.isNotBlank(formFieldPath)) {
+			nsAndPathTemp = namespace + FORM_NAMESPACE_PATH_SEPARATOR + formFieldPath;
+		} else if (StringUtils.isNotBlank(namespace)) {
+			nsAndPathTemp = namespace + FORM_NAMESPACE_PATH_SEPARATOR;
+		} else if (StringUtils.isNotBlank(formFieldPath)) {
+			nsAndPathTemp = FORM_NAMESPACE_PATH_SEPARATOR + formFieldPath;
+		}
+
+		if (nsAndPathTemp.length() > FORM_NAMESPACE_PATH_MAX_LENGTH) {
+			throw new APIException("Obs.namespaceAndPathTooLong", (Object[]) null);
+		}
+		if (StringUtils.countMatches(nsAndPathTemp, FORM_NAMESPACE_PATH_SEPARATOR) > 1) {
+			throw new APIException("Obs.namespaceAndPathNotContainSeparator", (Object[]) null);
+		}
+
+		markAsDirty(this.formNamespaceAndPath, nsAndPathTemp);
+		formNamespaceAndPath = nsAndPathTemp;
+	}
+
+	protected abstract void markAsDirty(Object oldValue, Object newValue);
```

Comment:
In light of the above you'll be able to remove this. The reason being that from recordable OpenMRS data and dirty-able OpenMRS data are two different things, I don't think we should conflate them there. @dkayiwa, @ibacher thoughts?

Reply:
I agree. We don't really use this pattern outside of Obs, so it's hard to see why it would be a requirement for the `formNamespaceAndPath` functionality.

Reply:
I also agree!

Reply:
Removed.